src/channel.rs

```diff
@@ -78,15 +78,16 @@ where
     where
         F: Fn() -> T + Send + Sync + 'static,
```

Comment:
```suggestion
        F: FnMut() -> T + Send + Sync + 'static,
```
Either this or have separate methods (`set` and `set_mut`) - doing that would also mean we should make a `DynamicMut` member of `ValueSource` - with `DynamicMut` having the `Mutex` and `Dynamic` not.

Reply:
Great idea, I'll try to do this as soon as I have some time.
Feel free to take over if I don't get back to you soon enough though.